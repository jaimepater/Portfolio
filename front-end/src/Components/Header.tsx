import * as React from 'react';
import { AppBar, IconButton, MenuList, Toolbar } from '@material-ui/core';
import MenuIcon from '@material-ui/icons/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import { Link, useLocation } from 'react-router-dom';
import styled from 'styled-components';
import Logo from './Logo';
import { getRoutes } from '../Commons/Utils/getRoutes';

const StyledMenuList = styled(MenuList)`
  display: flex;
`;

const StyledToolbar = styled(Toolbar)`
  justify-content: space-between;
  align-items: center;
`;

export const StyledMenuIcon = styled.div`
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex: 1;
`;
export const StyledLanguage = styled.div`
  flex: 1;
`;

const Header = () => {
  const { pathname } = useLocation();
  return (
    <AppBar position="static">
      <StyledToolbar>
        <StyledMenuIcon>
          <IconButton edge="start" color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton>
          <Logo />
        </StyledMenuIcon>
        <StyledMenuList>
          {getRoutes().map(route => (
            <MenuItem
              component={Link}
              key={route.id}
              to={route.path}
              selected={route.path === pathname}
            >
              {route.name}
            </MenuItem>
          ))}
        </StyledMenuList>
        <StyledLanguage />
      </StyledToolbar>
    </AppBar>
  );
};

export default Header;
