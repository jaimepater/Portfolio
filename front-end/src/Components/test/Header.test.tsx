import React from "react";
import { shallow } from 'enzyme';
import Header from "../Header";


describe('Header', () => {
  it('Renders Header', () => {
    const shallowComponent = shallow(<Header  />);
    expect(shallowComponent).toMatchSnapshot();
  });
});