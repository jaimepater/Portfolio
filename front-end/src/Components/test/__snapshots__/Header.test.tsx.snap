// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Renders Header 1`] = `
<div>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        color="inherit"
        edge="start"
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <Logo />
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
